utils/db/users.ts

```diff
@@ -250,4 +250,24 @@ export const getAuthInfoFromDb = async function (user_id: string): Promise<AuthI
 			return { authInfo: null };
 		});
 	return authinfo_promise.authInfo;
-}
\ No newline at end of file
+}
+
+export const getUserIdByTopicName = async function(topic_name: string) {
+	const query = `SELECT id FROM users WHERE topic_name = $1`;
+	const params = [topic_name];
+  
+	const queryResult = await conn.query(query, params).catch((err) => {
+		console.error(`[getUserIdByTopicName] Query failed: Select from users where topic_name = ${topic_name}`, { pg_query: query }, err);
+    });
+
+	if (queryResult && queryResult.rowCount === 0) {
+		console.error(`[getUserIdByTopicName] No user found for topicName ${topic_name}`);
+		return null;
+	} else if (queryResult && queryResult.rowCount > 1) {
+		console.error(`[getUserIdByTopicName] Multiple users found for topicName ${topic_name}`);
+		return null;
```

Comment:
You can return the 0th element in this case as well (and add a TODO to handle this later). 
We have done the name in the `[...nextauth.js]` for handling auth as well.